udf/extract_canonical_glean_app_info.sql

```diff
@@ -0,0 +1,71 @@
+/*
```

Comment:
Should this UDF live inside of mozfun instead?

Reply:
I would indeed like to get it into `mozfun` eventually, but I see a few blockers:

- It depends on `udf.fenix_build_to_datetime`, which we'd also need to move to mozfun
- It currently only supports Fenix
- This UDF still feels a bit volatile; it's not clear that the two arguments we have in here now are going to be sufficient long-term to disambiguate app names and channels

So I'm thinking of this a bit as an incubation stage before committing to an interface and a long-term name in mozfun. And I'm definitely open to suggestions on naming.

Reply:
It seems like it would be better to just commit with a single location for this UDF, especially if it's intended to be used across projects. FWIW, it would be useful to have `fenix_build_to_datetime` in mozfun too.